devauth/devauth.go

```diff
@@ -1076,23 +1070,22 @@ func (d *DevAuth) PreauthorizeDevice(ctx context.Context, req *model.PreAuthReq)
 }
 
 func (d *DevAuth) RevokeToken(ctx context.Context, tokenID string) error {
-
 	l := log.FromContext(ctx)
 	tokenOID := oid.FromString(tokenID)
 
-	if d.cache != nil {
-		token, err := d.db.GetToken(ctx, tokenOID)
+	l.Warnf("Revoke token with jti: %s", tokenID)
+	err := d.db.DeleteToken(ctx, tokenOID)
+
+	if err == nil && d.cache != nil {
+		var token *jwt.Token
+		token, err = d.db.GetToken(ctx, tokenOID)
```

Comment:
@tranchitella I'm missing something here:

aren't we guaranteed to not get a token here (right after `err := d.db.DeleteToken(ctx, tokenOID)`, and never proceed to actually clearing the cache?

could you expand a bit on where the race condition occurs?

Reply:
right; looks like we should get the token from db first, remove it from db and clear the cache

Reply:
You are right, sorry, dumb mistake.